models/user.go

```diff
@@ -77,11 +78,24 @@ func (db *DB) CreateUser(user *User) (int64, error) {
 	return userID, err
 }
 
-/*
-func (db *DB) ReadUser(user User) ([]*User, error) {
+//ReadUser returns one user from the database given userID
+func (db *DB) ReadUser(userID int64) (*User, error) {
```

Comment:
I think this should have a `columns` parameter, so you can just pass the `includes` array from the handler. Then you just build the query string with the `columns` array and if the `columns` array is nil, just make it an `*`, so it returns all fields by default. This way in the handler, you don't have to rebuild the `responseUser` from the returned user and just set the `Password` field of the returned user to an empty string before responding with it.

Reply:
How important is this change? I haven't found a solution for this.

Reply:
It's not super important. I just thought it might be cleaner, but you definitely don't have to do it. Just for reference though, this would essentially be the solution:
```
func (db *DB) ReadUser(userID int64, columns ...string) (*User, error) {
	columnStr := ""
	user := &User{}
	row := []interface{}{}
	if len(columns) == 0 {
		columnStr = "*"
		row = append(row, &(user.ID), &(user.Name), &(user.Email),
			&(user.Password), &(user.AvatarURL))
	} else {
		for _, column := range columns {
			switch column {
			case "ID":
				row = append(row, &(user.ID))
			case "Name":
				row = append(row, &(user.Name))
			case "Email":
				row = append(row, &(user.Email))
			case "AvatarURL":
				row = append(row, &(user.AvatarURL))
			default:
				return nil, errors.New("Invalid column name")
			}
		}
		columnStr = strings.Join(columns, ",")
	}
	queryString := fmt.Sprintf("SELECT %s FROM %s WHERE ID=?", columnStr, usersTable)

	err := db.QueryRow(queryString, userID).Scan(row...)
```